cgnsutilities/cgnsutilities.py

```diff
@@ -2288,9 +2314,18 @@ def addBocoDataSet(self, bocoDataSet):
 
     def coarsen(self, direction):
         """Coarsen the range of the BC along the specified direction"""
+
         for j in range(2):
             self.ptRange[direction, j] = (self.ptRange[direction, j] - 1) // 2 + 1
 
+        # coarsen the data set if it is an array
+        if self.dataSets:
+            for data_set in self.dataSets:
+                for dir_arr in data_set.dirichletArrays:
+                    dir_arr.dataDimensions[0] = numpy.prod(self.ptRange[:, 1])
+                    data_size = numpy.prod(dir_arr.dataDimensions)
+                    dir_arr.dataArr = dir_arr.dataArr[:data_size]
```

Comment:
So this will just truncate the data, not take every other element?

Reply:
That is right. The data is stored in a 1-D array, at least for all our meshes. I have added those assumptions as comments.

Reply:
I could not see the comment you mentioned btw. If we go with the truncate option we can either add it in the docstring of the coarsen method and maybe even add a warning if we do it.

Would it be too difficult to also take every other element (or even do some sort of restriction operation) instead of truncating it? I would expect for most use cases of the array BCs, you would set them from the python script anyways, but doing it like this looks like its digging a rabbit hole for an incoming student picking up a mesh with an array BC, coarsening it, and then that changing the entire BC value. I get that it may become difficult figuring out the 1d array indexing but it looks like it would be worth the effort if we do it now rather than someone else later.

Reply:
Alright,  I updated it to take every other element.